cluster/clustering/__init__.py

```diff
@@ -7,23 +7,24 @@
 from .clusterer import DBScanClusterer
 
 INDICO_VALUES = ["keywords", "title_keywords", "people", "places", "oragnizations"]
+BAD_ENTITIES = ["Computers & Peripherals", "EXCLUSIVE"]
 
 def generate_clusters_dict(entries, all_clusters, all_similarities, feature_vectors):
     # Build Result Dict
     count = defaultdict(int)
     cluster_features = {}
     result_dict = defaultdict(lambda: defaultdict(lambda: defaultdict(int)))
     for i in xrange(len(all_clusters)):
-        if all_clusters[i] < 0:
+        if all_clusters[i] < 0 or count[all_clusters[i]] > 20:
             continue
         count[all_clusters[i]] += 1
-        if count[all_clusters[i]] > 20:
-            continue
         entry, cluster = entries[i], all_clusters[i]
         entry['cluster'], entry["distance"] = cluster, all_similarities[i]
         cluster_features[cluster] = cluster_features.get(cluster, []) + [feature_vectors[i]]
         result_dict[cluster]["articles"] = result_dict[cluster].get("articles", []) + [entry]
 
+    [result_dict.pop(cluster_num) for cluster_num, cluster in result_dict.items() if len(cluster.get('articles')) < 2]

```

Comment:
Hm, maybe something like
`result_dict = {cluster_num: cluster for cluster_num, cluster in result_dict.iteritems() if len(cluster.get("articles")) >= 2}`
might be a bit more intuitive to read.